WelsonJS.Toolkit/WelsonJS.Esent/Schema.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Schema.cs (WelsonJS.Esent)
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2025 Namhyeon Go, Catswords OSS and WelsonJS Contributors
+// https://github.com/gnh1201/welsonjs
+// 
+using System;
+using System.Collections.Generic;
+
+namespace WelsonJS.Esent
+{
+    public class Schema
+    {
+        public string TableName { get; set; }
+        public List<Column> Columns { get; set; }
+        public Column PrimaryKey
+        {
+            get
+            {
+                return Columns.Find(c => c.IsPrimaryKey) ?? null;
+            }
+        }
+
+        public Schema(string tableName, List<Column> columns)
+        {
+            TableName = tableName;
+            Columns = columns ?? new List<Column>();
+        }
+
+        public void SetPrimaryKey(string columnName)
+        {
+            Column column = Columns.Find(c => c.Name.Equals(columnName, StringComparison.OrdinalIgnoreCase));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential NoSQL injection risk detected. Untrusted input is being used directly in database queries without proper validation or sanitization. This can allow attackers to manipulate queries, leading to unauthorized data access, modification, or deletion. To mitigate this risk, always validate and sanitize user input, use parameterized queries or prepared statements. Learn more - https://cwe.mitre.org/data/definitions/943.html